### PR TITLE
Merge changes from r1.5 into master

### PR DESCRIPTION
**Pull request by angerson**
This change picks up the commits exclusive to the r1.5 branch and puts them back into master.

There were a bunch of merge conflicts here. I favored master in most cases except those to do with obvious versioning differences.

I'm not sure if I did the merge correctly, considering there are a great many CLs presented here.

### CODE REVIEW

File: tensorflow/core/ops/compat/ops_history.v1.pbtxt [hunk 1]
```diff
@@ -34387,6 +34387,14 @@ op {
     name: "max_b"
     type: DT_FLOAT
   }
+  input_arg {
```
Comment (av8ramit) [inline tensorflow/core/ops/compat/ops_history.v1.pbtxt]: Can you check if this should be merged in?

File: tensorflow/tools/ci_build/Dockerfile.pi [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:14.04
```
Comment (av8ramit) [inline tensorflow/tools/ci_build/Dockerfile.pi]: Check if this was reverted in master?

File: tensorflow/tools/ci_build/Dockerfile.pi-python3 [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:14.04
+FROM ubuntu:16.04
```
Comment (av8ramit) [inline tensorflow/tools/ci_build/Dockerfile.pi-python3]: Same as above.

File: tensorflow/workspace.bzl [hunk 4]
```diff
@@ -115,11 +115,16 @@ def tf_workspace(path_prefix="", tf_repo_name=""):
   tf_http_archive(
       name = "eigen_archive",
       urls = [
-          "https://mirror.bazel.build/bitbucket.org/eigen/eigen/get/14e1418fcf12.tar.gz",
-          "https://bitbucket.org/eigen/eigen/get/14e1418fcf12.tar.gz",
+          "https://mirror.bazel.build/bitbucket.org/eigen/eigen/get/034b6c3e1017.tar.gz",
```
Comment (av8ramit) [inline tensorflow/workspace.bzl]: Check which version is later, master or r1.5 and keep the appropriate one.

**Review by av8ramit — CHANGES_REQUESTED** [review]

**Comment by angersson** [comment]
I addressed @av8ramit's changes after doing some hearty investigation as to why they happened.

File: tensorflow/python/eager/tensor_test.py [hunk 5]
```diff
@@ -174,7 +174,7 @@ def testMultiLineTensorRepr(self):
     self.assertIn("id=%d, shape=%s, dtype=%s, numpy=\n%r" %
                   (t._id, t.shape, t.dtype.name, t.numpy()), tensor_repr)
 
-  def testTensorStrReprObeyNumpyPrintOptions(self):
```
Comment (av8ramit) [inline tensorflow/python/eager/tensor_test.py]: Is keeping this change the right move?

**Review by av8ramit — APPROVED** [review]

**Comment by angersson** [comment]
Re-un-disabled some tests that were removed by a merge.

**Pull request merged**